src/main/kotlin/tech/harmonysoft/oss/gradle/dist/BuildCustomGradleDistributionTask.kt

```diff
@@ -13,27 +13,55 @@ import java.nio.file.Files
 import java.util.Properties
 import java.util.Stack
 import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.provider.Property
+import org.gradle.api.tasks.InputDirectory
 import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.TaskAction
 import org.gradle.tooling.BuildException
 import tech.harmonysoft.oss.gradle.dist.config.CustomGradleDistConfig
 
+@Suppress("LeakingThis")
 abstract class BuildCustomGradleDistributionTask : DefaultTask() {
 
     @get:Internal
     abstract val config: Property<CustomGradleDistConfig>
 
-    private val includeRootDir: File
-        get() = project.file("src/main/resources/include")
+    @get:Internal
+    abstract val downloadRootDir: DirectoryProperty
```

Comment:
do we still need this property? It's used only in tests

Reply:
Yes, I use this in my multiproject, where the project structure is as below, and I have this property configured so that gradle distributions are downloaded to the `build/download` directory in the root project. This makes rebuilding individual custom distributions much faster.

```
build/
dists/
    8.2.1-bin/
        build/
        src/
    8.2.1-all/
        build/
        src/
```